### ISSUE TITLE
improve image quality of cropped jpg files by disabling chroma subsampling

### ISSUE THREAD

**Issue by LaserBorg** [ultralytics/yolov5#7007]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

opencv was used to save ROI crops, but imwrite is limited at jpg quality since it uses  2x2 chroma subsampling, often reffered to as "4:2:2" (-> jpg is YUV color space) even when setting cv2.IMWRITE_JPEG_QUALITY to 100. 
Quality loss is clearly visible when working on images with highly saturated colors or sharp color changes.

I propose using pillow to write jpg files, with subsampling flag set to False. It is already in the requirements and the existing numpy matrix can easily get converted to pil format.

### Use case

image quality of the crops is crucial when cropped output files are further used as training data, while jpg compression still offers great benefit-cost ratio in filesize and compression speed compared to lossless (png) or even uncompressed (bmp) formats.

### Additional

I have a PR ready.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@LaserBorg good news 😃! Your original issue may now be fixed ✅ in PR #7008. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by Mob-Nikhil898** [comment]
The current save_one_box() is not saving anything in the desired folder.

**Comment by glenn-jocher** [comment]
@Mob-Nikhil898 saving crops works correctly.

<img width="1423" alt="Screen Shot 2022-05-08 at 12 47 56 PM" src="https://user-images.githubusercontent.com/26833433/167294753-f2e7d0c7-8257-41f9-8b40-ff8ddb3b7d12.png">

We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by Mob-Nikhil898** [comment]
Thank you bro, I messed up my detect.py settings that's why it was not saving .  My bad.

**Comment by glenn-jocher** [comment]
@Mob-Nikhil898 no worries at all! It happens to the best of us 😊. If you encounter any other issues or have further questions, feel free to ask. Happy to help anytime!
